src/viewProviders/topics.ts

```diff
@@ -86,49 +94,92 @@ export class TopicViewProvider implements vscode.TreeDataProvider<TopicViewProvi
     setContextValue(ContextValues.kafkaClusterSelected, false);
     this.kafkaCluster = null;
     this.treeView.description = "";
+    this.setSearch(null);
     this.refresh();
   }
 
   getTreeItem(element: TopicViewProviderData): vscode.TreeItem | Thenable<vscode.TreeItem> {
-    if (element instanceof Schema) {
-      return new SchemaTreeItem(element);
-    } else if (element instanceof KafkaTopic) {
-      return new KafkaTopicTreeItem(element);
+    let treeItem: vscode.TreeItem;
+    if (element instanceof KafkaTopic) {
+      treeItem = new KafkaTopicTreeItem(element);
+    } else if (element instanceof Schema) {
+      treeItem = new SchemaTreeItem(element);
+    } else {
+      // should only be left with ContainerTreeItems
+      treeItem = element;
+    }
+
+    if (this.itemSearchString) {
+      if (itemMatchesSearch(element, this.itemSearchString)) {
+        // special URI scheme to decorate the tree item with a dot to the right of the label,
+        // and color the label, description, and decoration so it stands out in the tree view
+        treeItem.resourceUri = vscode.Uri.parse(
+          `${SEARCH_DECORATION_URI_SCHEME}:/${element.searchableText()}`,
+        );
+      }
+      treeItem = updateCollapsibleStateFromSearch(element, treeItem, this.itemSearchString);
     }
-    return element;
+
+    return treeItem;
   }
 
   async getChildren(element?: TopicViewProviderData): Promise<TopicViewProviderData[]> {
-    let topicItems: TopicViewProviderData[] = [];
+    let children: TopicViewProviderData[] = [];
 
     if (element) {
       // --- CHILDREN OF TREE BRANCHES ---
       // NOTE: we end up here when expanding a (collapsed) treeItem
-      if (element instanceof ContainerTreeItem) {
-        // Local / CCloud containers, just return the topic tree items
-        return element.children;
-      } else if (element instanceof KafkaTopic) {
-        return await loadTopicSchemas(element);
+      if (element instanceof KafkaTopic) {
+        // return schema-subject containers
+        children = await loadTopicSchemas(element);
+      } else if (element instanceof ContainerTreeItem) {
+        // schema-subject container, return schema versions for the topic
+        children = element.children;
```

Comment:
Cosmetic change; I was annoyed that the previous order was the reverse of what we would actually see in the UI.